### PR TITLE
Sprint1 : combine PR for visualization from Sprint1

### PR DESCRIPTION
**Pull request by abasatwar**
### Description
Visualization implementatationfor Time series, Heatmap, Pie, Table view and UI chnages for config panel is included in this PR.

**Issues Resolved**
[Enhancement]: UI for dimension and metrics #797
[Enhancement]: Time Series - Visualizations config panel features v1 #794
[ENHANCEMENT]: Heatmap visualization config panel features v1 #791
[ENHANCEMENT]: Pie visualization config panel features v1 - #793
Table View Enhancement for Visualization #772

**Check List**
 - [x] New functionality includes testing.
- [x] Commits are signed per the DCO using --signoff
 
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx [hunk 1]
```diff
@@ -131,16 +142,16 @@ export const DataTable = ({ visualizations, layout, config }: any) => {
...
-  }, [isFullScreen]);
+  }, []);
 
   const hideGridFullScreenHandler = (e: any) => {
     if (e.key === 'Escape') {
-      if (isFullScreen) {
-        setIsFullScreen(false);
-      }
+      setIsFullScreen(false);
+      // if (isFullScreen) {
```
Comment (mengweieric) [inline dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx]: Please remove commented code here

**Comment by mengweieric** [comment]
Please use common PR template for this and moving forward. Add briefing for changes that have been made, and check items that apply. [Sample PR](https://github.com/opensearch-project/observability/pull/763)

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        <EuiButtonGroup
+          legend={legend}
+          options={options}
+          idSelected={idSelected}
+          onChange={onChange}
+        />
+      </>)
+  }
+  console.log("schemas:", schemas);
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx]: It is already been deleted in later commit and not present in PR.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx [hunk 3]
```diff
@@ -43,42 +29,77 @@ export const ConfigGraphStyle = ({
...
-  const handleChange = (id) => {}
-  const toggleStyeButtons = schemas[0].props.options.map((btn: any) => ({ id: btn.modeId, label: btn.name }));
+  const [styleGroup, interpolationGroup, lineWidth, fillOpacity] = schemas;
 
- 
+  const [sliderRange, setValue] = useState({
+    lineWidth: lineWidth.defaultState,
+    opacity: fillOpacity.defaultState
+  });
+  const onChange = (e) => {
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx]: This code is already updated in later commit.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx [hunk 4]
```diff
@@ -43,42 +29,77 @@ export const ConfigGraphStyle = ({
...
+      return (
+        <>
+          <DimensionComponent key={`viz-series-${index}`} {...params} />
+          <EuiSpacer size="s" />
+        </>
+      );
+    });
+  }, [schemas, vizState, handleConfigurationChange])
 
+  console.log("visualization:: ", visualizations)
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_graph_style.tsx]: This code is already updated in later commit.

File: dashboards-observability/public/components/visualizations/charts/lines/line.tsx [hunk 5]
```diff
@@ -25,10 +25,12 @@ export const Line = ({ visualizations, layout, config }: any) => {
...
     dataConfig?.valueOptions && dataConfig.valueOptions.xaxis ? dataConfig.valueOptions.yaxis : [];
   const lastIndex = fields.length - 1;
-  const mode =
-    dataConfig?.chartOptions && dataConfig.chartOptions.mode && dataConfig.chartOptions.mode[0]
-      ? dataConfig.chartOptions.mode[0].modeId
-      : 'line';
+
+  const mode = dataConfig?.graphStyle?.style
+    ? dataConfig.graphStyle.style
+    : 'lines';
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/visualizations/charts/lines/line.tsx]: This code is already updated in later commit and strings are moved to constants file.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 6]
```diff
@@ -57,30 +57,32 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations, tabID }:
     let list = { ...configList };
     let listItem = { ...list[name][index] };
     listItem = { ...listItem, [field]: value };
-    const newList = {
+    const x = {
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: This code is already updated in later commit.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 7]
```diff
@@ -114,97 +96,89 @@ export const DataConfigPanelItem = ({ fieldOptionList, visualizations, tabID }:
...
+            </EuiFormRow>
+
+            <EuiFormRow
+              label="Custom label"
+            >
+              <EuiFieldText
+                placeholder="Custom label"
+                value={singleField.custom_label}
+                onChange={(e) => updateList(e.target.value, index, sectionName, 'custom_label')}
+                aria-label="Use aria labels when no actual label is in use" />
```
[the remark this answers is not in the document]
Comment (abasatwar) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: This is handled in later commit.

File: dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx (hunk 1 above)
Comment (abasatwar) [reply dashboards-observability/public/components/visualizations/charts/data_table/data_table.tsx]: This has been removed in later commit.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    } else if (vizState?.style === "bar") {
+      return schemas.filter((schema: IConfigPanelOptionSection) => !["interpolation", "pointSize"].includes(schema.mapTo));
+    } else if (vizState?.style === "markers") {
+      return schemas.filter((schema: IConfigPanelOptionSection) => ["style", "pointSize"].includes(schema.mapTo));
+    } else return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'interpolation');
+  }, [vizState]);
+
+  const dimensions = useMemo(() =>
+    currentSchemas && currentSchemas.map((schema: IConfigPanelOptionSection, index: string) => {
+      let params;
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx]: Thanks @mengweieric , Initialized params with default values.

**Comment by abasatwar** [comment]
> Please use common PR template for this and moving forward. Add briefing for changes that have been made, and check items that apply. [Sample PR](https://github.com/opensearch-project/observability/pull/763)

Hi @mengweieric , Update PR as per suggested PR tempate. Also , we have addressed all your comments.
It would be great if you can provide your inputs.

Thanks,
Amruta

**Review by mengweieric — DISMISSED** [review]

File: dashboards-observability/common/constants/data_table.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+export const GRID_PAGE_RANGE_DISPLAY = 5;
+export const COLUMN_DEFAULT_MIN_WIDTH = 100;
+export const GRID_PAGE_SIZES = [10, 50, 100];
+export const ROW_DENSITIES = [
+  { icon: 'tableDensityExpanded', height: 55, selected: false },
+  { icon: 'tableDensityNormal', height: 45, selected: false },
+  { icon: 'tableDensityCompact', height: 35, selected: true },
+];
+
+export const HEADER_HEIGHT = 35;
```
Comment (ps48) [inline dashboards-observability/common/constants/data_table.ts:16]: If the data table is used inside event analytics let's move this to `constants/explorer.ts`, if shared by other components let's move to `constants/shard.ts`

File: dashboards-observability/package.json [hunk 10]
```diff
@@ -17,8 +17,11 @@
     "@nteract/outputs": "^3.0.11",
     "@nteract/presentational-components": "^3.4.3",
     "@reduxjs/toolkit": "^1.6.1",
+    "ag-grid-community": "^27.3.0",
+    "ag-grid-react": "^27.3.0",
```
Comment (ps48) [inline dashboards-observability/package.json:21]: Why do we need to add this new dependency? Is there an issue where we discussed what was not available in EUI components. Adding more packages, puts lot more maintenance issues in long term.  Thus, we should have strong reasons to support addition of this package.

File: dashboards-observability/.cypress/utils/event_constants.js [hunk 11]
```diff
@@ -110,3 +110,31 @@ export const renderPieChart = () => {
     cy.get('[data-test-subj="comboBoxToggleListButton"]').eq(0).click();
     cy.get('[data-test-subj="comboBoxInput"]').eq(2).click();
 };
+
+export const renderDataConfig = () => {
+  cy.get('.euiResizablePanel.euiResizablePanel--middle').contains('Data Cofigurations');
```
[the remark this answers is not in the document]
Comment (pratibhapandey16) [reply dashboards-observability/.cypress/utils/event_constants.js:115]: Sure. Going forward we will make sure of using `data-test-subj` wherever application in place of classname.

File: dashboards-observability/package.json (hunk 10 above)
Comment (abasatwar) [reply dashboards-observability/package.json:21]: hi @ps48  ,
We have changed table view implementation as per [https://github.com/opensearch-project/observability/issues/772](url).And as per new implementation, new dependencies are added and we have confirmed it on call and tkt too.

thanks,
Amruta

**Review by mengweieric — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

**Pull request merged**